src/grpc/request.rs

```diff
@@ -18,13 +23,190 @@ pub fn create_grpc_request(url: Url, headers: HeaderMap, body: Vec<u8>) -> Reque
 pub async fn execute_grpc_request(
     runtime: &TargetRuntime,
     operation: &ProtobufOperation,
+    protobuf_set: &ProtobufSet,
     request: Request,
 ) -> Result<Response<async_graphql::Value>> {
     let response = runtime.http2_only.execute(request).await?;
 
+    let grpc_status = response
+        .headers
+        .get(GRPC_STATUS)
+        .and_then(|header_value| header_value.to_str().ok());
+
     if response.status.is_success() {
-        return response.to_grpc_value(operation);
+        return if grpc_status.is_none() || grpc_status == Some("0") {
+            response.to_grpc_value(operation)
+        } else {
+            Err(response.to_grpc_error(protobuf_set))
+        };
+    }
```

Comment:
do we still need to parse header manually or could parse tonic::Status and pass it instead? or handle it completely on response side?

Reply:
I think yes, because we need to just check one header, instead of full scope.